test/test_model.py

```diff
@@ -212,6 +213,26 @@ def check_remote_units():
             ['relation-list', '-r', '4', '--format=json'],
         ])
 
+    def test_is_leader_refresh(self):
+        self.backend = op.model.ModelBackend()
+        self.model = op.model.Model('myapp/0', ['db0', 'db1', 'db2'], self.backend)
+
+        # A sanity check.
+        self.assertGreater(time.monotonic(), op.model.ModelBackend.LEASE_REFRESH_PERIOD.total_seconds())
+
+        fake_script(self, 'is-leader', 'echo true')
+        self.assertTrue(self.model.unit.is_leader())
+
```

Comment:
Don't you want a
```
# Change the return value, but see that we use the cached value
# In practice, we know the leadership won't change before we
# would timeout and check again.
fake_script(self, 'is-leader', 'echo false')
self.assertTrue(self.model.unit.is_leader())
```
Otherwise, while you check that setting _leader_check_time = 0 *does* trigger a refresh, nothing in this test would fail if we just always refreshed. (eg, take out the caching, and I believe this test still passes)

Reply:
Good call.